m4/compiler.m4

```diff
@@ -6,22 +6,40 @@ AC_DEFUN([LIBMESH_SET_COMPILERS],
   # look for a decent C++ compiler or honor --with-cxx=...
   CXX_TRY_LIST="g++ icpc icc pgCC c++"
 
-     # -------------------------------------------------------------------
-     # MPI -- enabled by default.  Check for it now so we can be somewhat
-     #                             smart about which compilers to look for
-     # -------------------------------------------------------------------
-     AC_ARG_ENABLE(mpi,
-                   AS_HELP_STRING([--disable-mpi],
-                                  [build without MPI message passing support]),
-                   [AS_CASE("${enableval}",
-                            [yes], [enablempi=yes],
-                            [no],  [enablempi=no],
-                            [AC_MSG_ERROR(bad value ${enableval} for --enable-mpi)])],
-                   [enablempi=yes])
-
   AS_IF([test "$enablempi" != no],
-        [CXX_TRY_LIST="mpicxx mpiCC mpicc $CXX_TRY_LIST"],
-        AC_MSG_RESULT([>>> Disabling MPI per user request <<<]))
+        [
+          dnl Ok we want to enable mpi. Let's think about what compilers
+          dnl we want to try.
+          AS_IF([test x"$MPI" != x],
+                [
+                  dnl only try to set CXX if the user hasn't specified one already
+                  AS_IF([test -z "$CXX"],
```

Comment:
Is this safe to do *before* the --with-cxx support below?

Reply:
No it's not safe. It wouldn't mess up the setting of CXX but it could lead to incorrectly setting other flags like `MPI_USING_WRAPPERS`